src/hotspot/os/bsd/os_perf_bsd.cpp

```diff
@@ -120,8 +120,17 @@ size_t length = sizeof(cpu_load_info);
     if (sysctl(mib, miblen, &cpu_load_info, &length, NULL, 0) == -1) {
       return OS_ERR;
     }
+#elif defined(__FreeBSD__)
+    size_t alllength = length * nProcs * sizeof(long);
+    long *allcpus = NEW_C_HEAP_ARRAY(long, alllength, mtInternal);
```

Comment:
The NEW_C_HEAP_ARRAY macro looks like this:
```
#define NEW_C_HEAP_ARRAY(type, size, memflags)\
  (type*) (AllocateHeap((size) * sizeof(type), memflags))
```
so the size passed to it is not in bytes but in terms of type. e.g. `nProcs * CPUSTATES` is correct here.

Reply:
Ah right I missed the sizeof bit...